hls4ml/templates/vivado/nnet_utils/nnet_recurrent.h

```diff
@@ -0,0 +1,799 @@
+
+//
+//    rfnoc-hls-neuralnet: Vivado HLS code for neural-net building blocks
+//
+//    Copyright (C) 2017 EJ Kreinar
+//
+//    This program is free software: you can redistribute it and/or modify
+//    it under the terms of the GNU General Public License as published by
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//
+//    This program is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU General Public License for more details.
+//
+//    You should have received a copy of the GNU General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+#ifndef NNET_RECURSIVE_H_
+#define NNET_RECURSIVE_H_
+
+#include "nnet_common.h"
+#include "nnet_activation.h"
+#include "nnet_dense.h"
+#include "hls_stream.h"
+
+
+namespace nnet {
+
+struct rnn_config
+{
+    // Internal data type definitions
+    typedef float state_t;
+    typedef float U_t;  // State x Input
+    typedef float W_t;  // State x State
+    typedef float V_t;  // Output x State
+
+    // Layer Sizes
+    static const unsigned n_in = 10;
+    static const unsigned n_out = 2;
+    static const unsigned n_state = 2;
+    static const unsigned activation_type = activ_relu;
+    static const unsigned table_size = 1024;
+
+    // Resource reuse info
+    static const unsigned io_type = io_parallel;
+    static const unsigned reuse_factor = 1;
+    static const bool store_weights_in_bram = false;
+};
+
+// Recusive Neural Network (RNN)
+// Resources:
+//  - http://www.wildml.com/2015/09/recurrent-neural-networks-tutorial-part-1-introduction-to-rnns/
+//  - https://github.com/pangolulu/rnn-from-scratch
+// Notes:
+//  - RNN naming conventions adopted from the above links
+//      - newstate = activation(U*input + W*state)
+//      - output   = V*newstate
+//  - If softmax is needed on output, perform *outside* this operations
+template<class data_T, class res_T, typename CONFIG_T, typename ACT_CONFIG_T>
+void simple_rnn(
```

Comment:
It looks like these `simple_rnn` and `simple_rnn_static` methods are not accessible through the hls4ml conversion, only the LSTM and GRU. Is that correct? If so, these implementations should be removed. Or otherwise, they should be made accessible through the conversion flow.

Reply:
Yeah, we can remove them as people hardly use basic RNN layers.